route_test.go

```diff
@@ -0,0 +1,66 @@
+package mux
+
+import (
+	"net/http"
+	"regexp"
+	"sync"
+	"testing"
+)
+
+var testNewRouterMu sync.Mutex
+var testHandler = http.NotFoundHandler()
+
+func BenchmarkNewRouter(b *testing.B) {
+	testNewRouterMu.Lock()
+	defer testNewRouterMu.Unlock()
+
+	// Set the RegexpCompileFunc to the default regexp.Compile.
+	RegexpCompileFunc = regexp.Compile
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		testNewRouter(b, testHandler)
+	}
+}
+
+func BenchmarkNewRouterRegexpFunc(b *testing.B) {
+	testNewRouterMu.Lock()
+	defer testNewRouterMu.Unlock()
+
+	// We preallocate the size to 8.
+	cache := make(map[string]*regexp.Regexp, 8)
+
+	// Override the RegexpCompileFunc to reuse compiled expressions
+	// from the `cache` map. Real world caches should have eviction
+	// policies or some sort of approach to limit memory use.
+	RegexpCompileFunc = func(expr string) (*regexp.Regexp, error) {
+		if regex, ok := cache[expr]; ok {
+			return regex, nil
+		}
+
+		regex, err := regexp.Compile(expr)
+		if err != nil {
+			return nil, err
+		}
+
+		cache[expr] = regex
+		return regex, nil
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		testNewRouter(b, testHandler)
+	}
+}
+
+func testNewRouter(tb testing.TB, handler http.Handler) {
```

Comment:
```suggestion
func testNewRouter(handler http.Handler) {
```

I don't see this param used anywhere, so I'm in favor of cleaning it up.

Reply:
Can we come up with an assertion here, assert on some value of Router, Route, Walk...?

Alternatively, I'd use `_ testing.TB` to make it clear that the parameter is (yet) unused, leaving it in place for assertions in the future.

Reply:
Seeing as these are all benchmark tests, I'm content with not asserting on anything at this time.